src/cuda/image_function_cuda.cu

```diff
@@ -239,6 +239,59 @@ namespace
         }
     }
 
+    __global__ void rotateCuda( const uint8_t * in, uint32_t rowSizeIn, uint32_t inHeight, float inXStart, float inYStart, 
+                                uint8_t * out, uint32_t rowSizeOut, uint32_t outHeight, 
+                                float cosAngle, float sinAngle)
+    {
+        uint32_t outX = blockDim.x * blockIdx.x + threadIdx.x,
+                 outY = blockDim.y * blockIdx.y + threadIdx.y;
+       
+        // If this thread isn't for a valid pixel in the output image, then do nothing. 
+
+        if ( outX >= rowSizeOut || outY >= outHeight ) 
```

Comment:
Please change this condition to opposite and put the code inside the condition. `return` is not recommended to use in GPU kernels as all threads expect to finish their work at the end of kernel's body. So the code would be like this:
```cpp
if ( outX < rowSizeOut && outY < outHeight ) {
    // your code below
}
```

Reply:
Okay, done.